crates/protocol/src/resource/mod.rs

```diff
@@ -27,7 +29,7 @@ pub struct ResourceMetadata {
 }
 
 /// A struct representing the structure and common functionality available to all Resources.
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
```

Comment:
out of curiosity, when did you realize you had to add the `PartialEq` traits for resources?

Reply:
Good question!

By default, structs aren't comparable. This means if I had two Resource structs previously: `resourceA` and `resourceB`, this line would fail to compile: 
`let is_equal = resourceA == resourceB;`

The reason why I needed to add this derive was that we added a unit test [here](https://github.com/TBD54566975/tbdex-rs/pull/31/files#diff-b8533a0025b39bd85c0009de28d5872538505b83ff30f855614644169e412a3cR132) to ensure that two Resources are equal to one another. Without this trait, the assertion line doesn't compile.

That might sound kinda insane, and a lot of things to remember. But Rust has REALLY good error messages for situations like this, and will tell you exactly what you need to do in order to get it working. For example, here's the error message you get when `PartialEq` is missing:
```
error[E0369]: binary operation `==` cannot be applied to type `resource::Resource<offering::OfferingData>`
   --> crates/protocol/src/resource/offering.rs:132:9
    |
132 |         assert_eq!(offering, parsed_offering)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         resource::Resource<offering::OfferingData>
    |         resource::Resource<offering::OfferingData>
    |
note: an implementation of `PartialEq` might be missing for `resource::Resource<offering::OfferingData>`
   --> crates/protocol/src/resource/mod.rs:34:1
    |
34  | pub struct Resource<T> {
    | ^^^^^^^^^^^^^^^^^^^^^^ must implement `PartialEq`
    = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `resource::Resource<offering::OfferingData>` with `#[derive(PartialEq)]`
   --> crates/protocol/src/resource/mod.rs:34:1
    |
34  + #[derive(PartialEq)]
35  | pub struct Resource<T> {
    |

```